app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt

```diff
@@ -99,6 +99,7 @@ class DefaultBrowserToolbarController(
     }
 
     override fun handleReaderModePressed(enabled: Boolean) {
+        activity.finishActionMode()
```

Comment:
This works but adds to the tight coupling of this Controller to HomeActivity.
(we should probably refactor out the activity reference altogether, but in another ticket)
Ideally the Controller should not know or care whether the activity actually has a `finishActionMode` method.

The easiest way to decouple the Controller from HomeActivity would be to have the Controller receive another callback in it's constructor, something like `onReaderModeChanged: () -> Unit` maybe? (It could also send a Boolean) so that any clients of this Controller could decide whether they are interested or not in this functionality, without the Controller having to know anything about them.

Reply:
Added callback `onReaderModeChanged: () -> Unit = {}` to `DefaultReaderModeController`'s constructor